Minesweeper/BoardGenerator.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections;
+
+namespace Minesweeper
+{
+    public static class BoardGenerator
+    {
+        public static bool[,] Generate(IMinePlacementGenerator generator, int numberOfRows, int numberOfColumns, float density)
+        {
+            if (numberOfRows < 1 || numberOfColumns < 1)
+                throw new ArgumentException("Invalid dimensions given.");
+
+            if (density < 0.2 || density > 0.6)
+                throw new ArgumentException("Invalid mine density given.");
+            
+            if (generator == null)
+                generator = new RandomMinePlacementGenerator(numberOfRows, numberOfColumns);
```

Comment:
This is probably not an ideal solution, because you are breaking dependency inversion here by tightly coupling this with concrete implementation. I honestly don't see a reason of having an interface IMinePlacementGenerator. I think the logic of generating mines should be inside this method and no interface would be needed.